libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -1045,26 +1075,28 @@ void RCOutput::set_output_mode(uint32_t mask, const enum output_mode mode)
     }
 #if HAL_WITH_IO_MCU
     if ((mode == MODE_PWM_ONESHOT ||
-         mode == MODE_PWM_ONESHOT125) &&
-        (mask & ((1U<<chan_offset)-1)) &&
-        AP_BoardConfig::io_enabled()) {
-        iomcu_mode = mode;
-        // also setup IO to use a 1Hz frequency, so we only get output
-        // when we trigger
-        iomcu.set_freq(io_fast_channel_mask, 1);
-        iomcu.set_oneshot_mode();
-        return;
-    }
-    if (mode == MODE_PWM_BRUSHED &&
+        mode == MODE_PWM_ONESHOT125 ||
+        mode == MODE_PWM_BRUSHED ||
+        (mode >= MODE_PWM_DSHOT150 && mode <= MODE_PWM_DSHOT600)) &&
         (mask & ((1U<<chan_offset)-1)) &&
         AP_BoardConfig::io_enabled()) {
         iomcu_mode = mode;
-        iomcu.set_brushed_mode();
+        iomcu.set_output_mode(mask, mode);
```

Comment:
need some testing of oneshot and brushed output with IOMCU to ensure we do get the right update rate

Reply:
oneshot and oneshot125 verified